test/e2e/basic_workflow_test.go

```diff
@@ -93,8 +104,18 @@ func (test *e2eTest) serviceList(t *testing.T, serviceName string) {
 	out, err := test.kn.RunWithOpts([]string{"service", "list", serviceName}, runOpts{NoNamespace: false})
 	assert.NilError(t, err)
 
-	expectedOutput := fmt.Sprintf("%s", serviceName)
-	assert.Check(t, util.ContainsAll(out, expectedOutput))
+	assert.Check(t, util.ContainsAll(out, serviceName))
+}
+
+func (test *e2eTest) serviceCreateDuplicate(t *testing.T, serviceName string) {
```

Comment:
IMO, this particular test case doesn't belong to `basic_workflow_test` as the filename suggest, but kind of testing for error handling mechanism, we should probably create another workflow which specifically aims at error handling test cases.

Reply:
If you all decide to move this to a more thorough IT (I am ambivalent at this point), please do the same for suggestion above on delete service.

Reply:
should be addressed as part of resolution to https://github.com/knative/client/issues/301